route/build/src/lake_route.f90

```diff
@@ -0,0 +1,137 @@
+ ! inout
+ TYPE(STRFLX), intent(inout), allocatable :: RCHFLX_out(:,:)   ! Reach fluxes (ensembles, space [reaches]) for decomposed domains
+ ! Output
+ integer(i4b), intent(out)                :: ierr           ! error code
+ character(*), intent(out)                :: message        ! error message
+ ! Local variables to
+ real(dp)                                 :: q_upstream     ! total discharge at top of the reach being processed
+ real(dp)                                 :: WB             ! water balance component in the lake
+ type(STRFLX), allocatable                :: fluxstate(:)   ! upstream Reach fluxes
+ INTEGER(I4B)                             :: nUps           ! number of upstream segment
+ INTEGER(I4B)                             :: iUps           ! upstream reach index
+ INTEGER(I4B)                             :: iRch_ups       ! index of upstream reach in NETOPO
+ character(len=strLen)                    :: cmessage       ! error message from subroutine
+
+ ! initialize error control
+ ierr=0; message='lake_route/'
+
+  ! identify number of upstream segments of the lake being processed
+  nUps = size(NETOPO_in(segIndex)%UREACHI)
+
+  allocate(fluxstate(nUps), stat=ierr, errmsg=cmessage)
+  if(ierr/=0)then; message=trim(message)//trim(cmessage)//': fluxstate'; return; endif
+
+  ! loop to get the streamlow of the usptream discharges to the lake
+  if (nUps>0) then
+    do iUps = 1,nUps
+      iRch_ups = NETOPO_in(segIndex)%UREACHI(iUps)      !  index of upstream of segIndex-th reach
+      fluxstate(iUps) = RCHFLX_out(iens,iRch_ups)
+    end do
+  endif
+
+  ! Find out total q draining into a lake
+  q_upstream = 0.0_dp
+  if(nUps>0)then
+   do iUps = 1,nUps
+    q_upstream = q_upstream + fluxstate(iUps)%REACH_Q_IRF
+   end do
+  endif
+
+  ! print*, "inside lake, RCHFLX_out(iens,segIndex)%basinprecip = ", RCHFLX_out(iens,segIndex)%basinprecip
+  ! print*, "inside lake, RCHFLX_out(iens,segIndex)%basinevapo = ", RCHFLX_out(iens,segIndex)%basinevapo
+  ! print*, "RPARAM_in(segIndex)%RATECVA", RPARAM_in(segIndex)%RATECVA
+  ! print*, "RPARAM_in(segIndex)%RATECVB", RPARAM_in(segIndex)%RATECVB
+  
+
+  ! perform lake routing based on a fixed storage discharge relationship Q=kS
+  ! no runoff input is added to the lake; the input are only precipitation and evaporation to the lake
+
+  print*, '------lake-simulation-------- '
+  print*, 'node id that is lake ...... = ', NETOPO_in(segIndex)%REACHID ! to check the reach id of lake
+  print*, 'lake param RATECVA ........ = ', RPARAM_in(segIndex)%RATECVA
+  print*, 'lake param RATECVB ........ = ', RPARAM_in(segIndex)%RATECVB
+  print*, 'volume before simulation m3 = ', RCHFLX_out(iens,segIndex)%REACH_VOL(0)
+  print*, 'upstream streamflow m3/s .. = ', RCHFLX_out(iens,segIndex)%REACH_Q_IRF
+  print*, 'upstream precipitation m3/s = ', RCHFLX_out(iens,segIndex)%basinprecip
+  print*, 'upstream evaporation m3/s . = ', RCHFLX_out(iens,segIndex)%basinevapo
```

Comment:
Yes, i think these could remain but could use ixDesire to enable spot check at a particular lake.  ixDesire is index of reach (here lake) for which this information is printed out.

Reply:
so basically I replace the "segIndex" with "ixDesire" but what if there is not ixDesire identified in the control file...
I guess I need a logical check first... if (ixDesire/=integerMissing) then ... correct?